amf-aml/shared/src/main/scala/amf/aml/client/scala/model/domain/DialectDomainElement.scala

```diff
@@ -15,68 +16,144 @@ case class DialectDomainElement(override val fields: Fields, annotations: Annota
     extends DomainElement
     with Linkable {
 
-  def objectCollectionProperty(f: Field): Seq[DialectDomainElement] = {
-    fields
-      .getValueAsOption(f)
-      .collect({
-        case Value(arr: AmfArray, _) =>
-          arr.values.collect({ case d: DialectDomainElement => d })
-      })
-      .getOrElse(Nil)
-  }
+  private[amf] def objectCollectionProperty(f: Field): Seq[DialectDomainElement] = getObjectByProperty(f.value.iri())
 
-  def objectProperty(f: Field): Option[DialectDomainElement] =
-    fields.getValueAsOption(f).collect({ case Value(d: DialectDomainElement, _) => d })
+  private[amf] def objectProperty(f: Field): Option[DialectDomainElement] =
+    objectCollectionProperty(f).headOption
 
-  def literalProperty(f: Field): Option[Any] =
-    fields.getValueAsOption(f).collect({ case Value(v: AmfScalar, _) => v.value })
+  private[amf] def literalProperty(f: Field): Option[Any] =
+    literalProperties(f).headOption
 
-  def literalProperties(f: Field): Seq[Any] =
-    fields
-      .getValueAsOption(f)
-      .collect({
-        case Value(v: AmfArray, _) =>
-          v.values.collect({ case s: AmfScalar => s.value })
-      })
-      .getOrElse(Nil)
+  private[amf] def literalProperties(f: Field): Seq[Any] = graph.scalarByField(f)
 
-  def fieldsToProperties: Iterable[(PropertyMapping, FieldEntry)] =
-    fields
-      .fields()
-      .flatMap(fe => propertyForField(fe.field).map(pm => (pm, fe)))
+  private[amf] override def componentId: String = ""
 
-  def propertyForField(f: Field): Option[PropertyMapping] =
+  private def setObjInCollection(f: Field, node: Either[YNode, YMapEntry], newObj: DialectDomainElement) = {
+    val annotations                     = annotationsFromEither(node)
+    val objs: Seq[DialectDomainElement] = fields.field(f)
+    set(f, AmfArray(objs :+ newObj), annotations)
+  }
+
+  private def annotationsFromEither(node: Either[YNode, YMapEntry]) =
+    node match {
+      case Left(value)  => Annotations(value)
+      case Right(value) => Annotations(value)
+    }
+
+  private def findPropertyMappingByTermPropertyId(termPropertyId: String): Option[PropertyMapping] =
     definedBy
       .propertiesMapping()
-      .find(_.nodePropertyMapping().option().contains(f.toString))
+      .find(_.nodePropertyMapping().value() == termPropertyId)
+
+  // Types of the instance
+  protected var instanceTypes: Seq[String] = Nil
```

Comment:
Why isn't instanceTypes part of the model?

Reply:
Because are the metadata, the values typically came from the node mapping